Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -10335,8 +10335,15 @@ bool WebPageProxy::didChooseFilesForOpenPanelWithImageTranscoding(const Vector<S
 {
 #if PLATFORM(MAC)
     auto transcodingMIMEType = WebCore::MIMETypeRegistry::preferredImageMIMETypeForEncoding(allowedMIMETypes, { });
-    if (transcodingMIMEType.isNull())
-        return false;
+
+    if (transcodingMIMEType.isNull()) {
+        // For designated sites which are sending "image/*", we need to force the mimetype
+        // to be able to transcode from HEIC to JPEG.
+        if (Quirks::shouldTranscodeHeicImagesForURL(currentURL()))
```

Comment:
The UIProcess side of things doesn't have a document object, so you have to check for the need to use quirks at the Page Settings level, like we do elsewhere in `WebPageProxy.cpp`.

Something like this:

```
        if (transcodingMIMEType.isNull() && protectedPreferences()->needsSiteSpecificQuirks()) {
            if (Quirks::shouldTranscodeHeicImagesForURL(currentURL()))
...
```

Reply:
I think the code needs to separate the two cases of isNull and needsSiteSpecificQuirks.

Reply:
New PR.